front/components/vaults/VaultCreateFolderModal.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  ExclamationCircleStrokeIcon,
+  Input,
+  Modal,
+  Page,
+  TextArea,
+} from "@dust-tt/sparkle";
+import type {
+  APIError,
+  DataSourceType,
+  VaultType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { isDataSourceNameValid } from "@dust-tt/types";
+import { useState } from "react";
+
+export default function VaultCreateFolderModal({
+  isOpen,
+  setOpen,
+  owner,
+  vault,
+  dataSources,
+}: {
+  isOpen: boolean;
+  setOpen: (isOpen: boolean) => void;
+  owner: WorkspaceType;
+  vault: VaultType;
+  dataSources: DataSourceType[];
+}) {
+  const [name, setName] = useState<string | null>(null);
+  const [nameError, setNameError] = useState<string | null>(null);
+
+  const [description, setDescription] = useState<string | null>(null);
+  const [descriptionError, setDescriptionError] = useState<string | null>(null);
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={() => {
+        setOpen(false);
+      }}
+      onSave={async () => {
+        if (!name) {
+          setNameError("Name is required.");
+          return;
+        }
+        const nameValidRes = isDataSourceNameValid(name);
+        if (nameValidRes.isErr()) {
+          setNameError(
+            "Name is invalid, must be multiple characters with no space."
+          );
+          return;
+        }
+        if (dataSources.find((ds) => ds.name === name)) {
```

Comment:
Not super scalable but that's what we currently have for website creation so I kept it!

Reply:
that test already avoid some errors, but ideally we should add a test in api call , something like a  DataSourceResource.fetchByName() before doing a makeNew - and send a 400 / datasource name already exist if it's found . In the end we have an index on name but we don't want to crash because of the constraint - if we can catch it before it's better.

Reply:
Agreed, keeping as is for now. 
-> I think we might want to revisit the constraint at some point, cause since you cannot access folders from another vault it might be a pain for user to not know which name they cannot use because already used in another vault they don't know about.

Reply:
right, we should get rid of this constraint - but since we still use datasource names in some places, we can't do it right now